testing/src/main/java/org/oppia/testing/FakeCrashLogger.kt

```diff
@@ -0,0 +1,30 @@
+package org.oppia.testing
+
+import java.lang.Exception
+import org.oppia.testing.TestLogReportingModule.Companion.exceptionList
+import org.oppia.util.firebase.CrashLogger
+
+/**
+ * A test specific fake crash logger that adds the logged exception to a list of exceptions.
+ */
+class FakeCrashLogger : CrashLogger {
+
+  override fun logException(exception: Exception) {
+    exceptionList.add(exception)
+  }
+
+  /** This is used to get the most recent exception present in the list of exceptions. */
+  fun getMostRecentException(): Exception {
+    val size = exceptionList.size
```

Comment:
A more Kotlin idiomatic way to do this would be:

```kotlin
fun getMostRecentException(): Exception = exceptionList.last()
```

See documentation: https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/last.html.